packages/next/build/babel/plugins/next-ssg-transform.ts

```diff
@@ -134,7 +134,19 @@ export default function nextTransformSsg({
     ident: NodePath<BabelTypes.Identifier>
   ): boolean {
     const b = ident.scope.getBinding(ident.node.name)
-    return b != null && b.referenced
+    if (b?.referenced) {
```

Comment:
```suggestion
    if (b?.referenced || !b?.constant) {
```

Reply:
Discussed we'd rather have it eliminate both, aggressively.

Reply:
ignore as per discussion.